personio-personnel-data-api-oa3.yaml

```diff
@@ -2139,6 +2140,146 @@ paths:
             problem+json:
               schema:
                 $ref: "#/components/schemas/ErrorResponse"
+    post:
+      security:
+        - BearerAuth: [ ]
+      tags:
+        - Absences
+      description: Adds absence data for absence types **tracked in hours**.
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              $ref: "#/components/schemas/CreateAbsencePeriodRequest"
+        description: Absence data
+        required: true
+      x-readme:
+        code-samples:
+          - language: "powershell"
+            code: "Auto-generated code snippets for this language may not be supported"
+          - language: "php"
+            code: "Auto-generated code snippets for this language may not be supported"
+      responses:
+        "201":
+          description: The absence period is successfully created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: "#/components/schemas/HourlyAbsencePeriodResponse"
+              examples:
+                response:
+                  value:
+                    success: true
+                    data:
```

Comment:
[Question]: Why is an array returned when only a single Absence Period is created?

Reply:
currently in the code implementation we have a generic response creation function which is used in multiple places doing different functionalities, this method also does some checks and sometimes return empty arrays under certain conditions, that's why we don't make any further modifications on the response object

NB:
1-  we can post process the output after doing some checks before sending it back to the response but till now we're keeping it in the generic way
2- we have the same response structure for the other daily/hourly get/post endpoints